src/dialects/postgres/query-generator.js

```diff
@@ -913,19 +1076,18 @@ export class PostgresQueryGenerator extends AbstractQueryGenerator {
    * @param {string} schemaName
    */
   getForeignKeyReferencesQuery(tableName, catalogName, schemaName) {
-    return `${this._getForeignKeyReferencesQueryPrefix()
-    }WHERE constraint_type = 'FOREIGN KEY' AND tc.table_name = '${tableName}'${
-      catalogName ? ` AND tc.table_catalog = '${catalogName}'` : ''
-    }${schemaName ? ` AND tc.table_schema = '${schemaName}'` : ''}`;
+    return this.getForeignKeysQuery({ tableName, schema: schemaName });
```

Comment:
This method is only used by `PostgresQueryInterface#getForeignKeyReferencesForTable`. It would be interesting to look into whether that method and this one can be removed

Reply:
There seems to be a lot of redundancy around constraints.  It's easier for me to think of retrieving foreignKeys and filtering through code from there.  I don't know about datawarehouses that might have tons of keys, but I don't think it'd be a large impact or much additional info to process locally.